controllers/feeds/getReaction.js

```diff
@@ -26,10 +26,16 @@ module.exports = async (req, res) => {
     const token = req.token;
     const detailFeed = await getPlainFeedById(feed_id);
     const myAnonymUser = await getAnonymUser(req.userId);
-    const karmaScores = await UsersFunction.getUsersKarmaScore(User, detailFeed.actor.id);
     getstreamService
       .getReaction(activity_id, token, limit)
       .then(async (result) => {
+        let actors = [];
+        result.results.map((data) => {
+          if (data.user_id) {
+            actors.push(data.user_id);
+          }
+        });
+        const karmaScores = await UsersFunction.getUsersKarmaScore(User, actors);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `Array.prototype.map` is not appropriate here as you're using it for its side effect (pushing to the `actors` array) rather than for its intended purpose (creating a new array based on an existing one). It would be more appropriate to use `Array.prototype.forEach` or `for...of` loop. Also, consider using `const` instead of `let` for `actors` since it's not being reassigned.

```diff
- let actors = [];
- result.results.map((data) => {
-   if (data.user_id) {
-     actors.push(data.user_id);
-   }
- });
+ const actors = [];
+ result.results.forEach((data) => {
+   if (data.user_id) {
+     actors.push(data.user_id);
+   }
+ });
```

Also, there's no error handling for the promise returned by `getstreamService.getReaction`. If this promise rejects, the error will be unhandled and could lead to unexpected behavior. Consider adding a `.catch` block to handle potential errors.

```diff
  getstreamService
    .getReaction(activity_id, token, limit)
    .then(async (result) => {
      // ...
    })
+   .catch(error => {
+     console.error('Error getting reaction:', error);
+     // Handle the error appropriately...
+   });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->